server/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java

```diff
@@ -1219,6 +1219,12 @@ public void testDataNodeRestartWithBusyMasterDuringSnapshot() throws Exception {
         disruption.startDisrupting();
         logger.info("-->  restarting data node, which should cause primary shards to be failed");
         internalCluster().restartNode(dataNode, InternalTestCluster.EMPTY_CALLBACK);
+
+        logger.info("-->  wait for shard snapshots to show as failed");
+        assertBusy(() -> assertThat(
```

Comment:
Before this change we would sometimes unblock the node and stop the disruption before the first shard failure. I think this makes the test weaker. I'm guessing it's invalid to do this after `disruption.stopDisrupting()`? If so, can we for instance only do it sometimes (with a comment saying why we don't always do it)?

Reply:
> Before this change we would sometimes unblock the node and stop the disruption before the first shard failure.

I'd argue that's a good thing :) <= The whole point of this test was to test this situation (failure on the data node before CS updates resume). The case where we stop disrupting before anything fails is probably practically impossible and even if it wasn't something that's covered in `SnapshotResiliencyTests` (where want want that kind of randomness because we can reproduce things) anyway.
WDYT?

Reply:
I'm not convinced yet. Practically impossible is not impossible enough for me :)

Do you think that the failure in #48526 is also captured, rarely, by the `SnapshotResiliencyTests`? Can we make a more focussed test there?

Reply:
The situation we're running into is perfectly covered by `SnapshotResiliencyTests` but those tests don't use the status API (which is the only thing that's functionally broken here) so they don't fail.
We could maybe add a test that involves the status APIs to `SnapshotResiliencyTests` to cover these things there as well if we want more randomness. I think that would be better than purposely making this test run into an, let's say almost impossible, situation at some point and not be reproducible anyway?

Reply:
I see, yes, `SnapshotResiliencyTests` sounds like a better place for this.

Reply:
I'd still rather adjust this test to have some reproducible testing for the concrete bug here and then enhance the `SnapshotResiliencyTests` down the line. That's basically what has been the strategy for other tests as well: make the ITs deterministic to reproduce known issues and and keep the randomness in `SnapshotResiliencyTests`.
Everything covered by this test is covered by `SnapshotResiliencyTests` anyway and this test was just added for `6.x` coverage, so I don't see us losing any coverage here by making this one more deterministic? :)